packages/client/hmi-client/src/services/goLLM.ts

```diff
@@ -0,0 +1,53 @@
+import API from '@/api/api';
+import useAuthStore from '@/stores/auth';
+import { TaskStatus } from '@/types/Types';
+import { logger } from '@/utils/logger';
+import { EventSourceMessage, fetchEventSource } from '@microsoft/fetch-event-source';
+
+export async function modelCard(documentId: string, modelId: string): Promise<any> {
```

Comment:
Add some JSDoc please.
And handle all the HTTP responses available for the endpoint.
Also add a try, catch.

Reply:
I think we need to refactor how we handle errors in responses, they are being intercepted and we are not able to handle the  response errors in the API call, instead we just get a null response.  I written the logic here for now, but I think this needs to be improved in its own ticket, happy to discuss.